serpentTools/parsers/results.py

```diff
@@ -1,15 +1,319 @@
+        self._resKeys = []  # Includes a complete set of .res variables
+        fObj = open(self.filePath, 'r')
+        numChar = 0  # count the number of characters
+        nUniv = len(self._univID)  # number of steps
+        nSteps = [self._univID[item] for item in self._univID][0]
+        for tline in fObj:
+            self._positionInFile(tline)
+            if numChar < self._ChunkPos[0]:
+                numChar += len(tline)
+                continue
+            # Not an empty line and not a comment
+            if tline.strip() != '' and not re.search(r'\%', tline):
+                currKey = re.search(r'^\w+', tline)
+                if currKey:
+                    keyName = convertVariableName(currKey.group())
+                    self._initResData(keyName, nSteps, tline)
+            numChar += len(tline)
+            if numChar >= self._ChunkPos[1]:
+                break
+        fObj.close()
+        self._idxSect = [False, False, False]
+
+    def _initResData(self, key, nSteps, tline):
+        """Initializes the dimensions of the variable"""
+        if key not in self._resKeys:
+            # Type of data (general meta, universes)
+            self._resKeys.append(key)
+            nvals = len(self._getDictVal(tline))
+            if self._idxSect[0]:
+                self.metadata[key] = ''
+            elif self._idxSect[1]:
+                self.resdata[key] = \
+                    empty((nSteps, nvals))
+
+    def _versionSerpent(self):
+        """Identifies the utilized version"""
+        with open(self.filePath, 'r') as fObject:
+            if fObject is None:
+                raise IOError("Attempting to read on file that does not exist.\n"
+                              "Parser: {}\nFile: {}".format(self, self.filePath))
+            for tline in fObject:
+                # If not an empty line or a comment
+                if re.search('VERSION', tline):
+                    version = self._getDictVal(tline)
+                    self._mapStrVersions(version)
+                    break
+
+    def _numUnivSteps(self):
+        """Identifies how many universes and burnup steps are included"""
+        #  Default number of universes and burnup steps
+        with open(self.filePath, 'r') as fObject:
+            for tline in fObject:
+                # If not an empty line or a comment
+                if re.search(self.strSearch['universe'], tline):
+                    currUniv = self._getDictVal(tline)
+                    self._addUniv(currUniv)
+
+    def _addUniv(self, currUniv):
+        """Add the current universe or add its counter"""
+        if currUniv in self._univID:
+            self._univID[currUniv] += 1
+        else:
+            self._univID[currUniv] = 1
+
+    def _getDictVal(self, tline):
+        """Gets the values to the dictionary"""
+        SepOptions = {
+            'list': r'\=.\[.+\]',
+            'singleVal': r'\=.+',
+            'string': r'\=.+\'.+\' '}
```

Comment:
If this is going to be used frequently, it would be better served to be added as a class attribute, that way we aren't recreating it every time we launch into this method.
```
class ResReader(XSReader):
    _SEP_OPTIONS = {-----}
```
This dictionary can then be accessed with `self._SEP_OPTIONS`